src/Bridges/Constraint/soc_to_nonconvex_quad.jl

```diff
@@ -0,0 +1,191 @@
+abstract type AbstractSOCtoNonConvexQuadBridge{T} <: AbstractBridge end
+
+"""
+    SOCtoNonConvexQuadBridge{T}
+
+Constraints of the form `VectorOfVariables`-in-`SecondOrderCone` can be
+transformed into a `ScalarQuadraticFunction`-in-`LessThan` and a
+`ScalarAffineFunction`-in-`GreaterThan`. Indeed, the definition of the
+second-order cone
+```math
+t \\ge \\lVert x \\rVert_2 \\  (1)
+```
+is equivalent to
+```math
+\\sum x_i^2 \\le t^2  (2)
+```
+with ``t \\ge 0``.  (3)
+
+*WARNING* This transformation starts from a convex constraint (1) and creates a
+non-convex constraint (2), because the Q matrix associated with the constraint 2
+has one negative eigenvalue. This might be wrongly interpreted by a solver.
+Some solvers can look at (2) and understand that it is a second order cone, but
+this is not a general rule.
+For these reasons this bridge is not automatically added to the LazyBridgeOptimizer.
```

Comment:
I would say "added by `full_bridge_optimizer`" instead. No bridge is added in the `LazyBridgeOptimizer` constructor.